app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :likes, foreign_key: 'postId'
+  has_many :comments, foreign_key: 'postId'
+  belongs_to :user, foreign_key: 'authorId'
+
+  def update_posts_counter
+    user.update(postsCounter: user.posts.count)
+  end
+
```

Comment:
- [ ] I think it would be better if you improve the `update_posts_counter` so it update the posts counter for a User when he has added a new one. For that i suggest you use the ``increment!`` method as the one you implemented, is set the attribute to be the count of associated comments, instead of implementing a method which increments the attribute by 1 for the created post.

- [ **OPTIONAL** ] **IMPORTANT:** I think it would be better if you use use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to database, then make the `update_post_counter` method privateStudents can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the `update_posts_counter` method as `private`.
You `Post` class should look to something like this:
```ruby
class Post < ApplicationRecord
  ....

  def five_most_recent_comments
    comments.order(created_at: :desc).limit(5)
  end

  after_save :update_posts_counter

  private

  def update_posts_counter
    author.increment!(:postsCounter)
  end
end
```